system-test/speech.js

```diff
@@ -0,0 +1,411 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var assert = require('assert');
+var async = require('async');
+var fs = require('fs');
+var path = require('path');
+var uuid = require('node-uuid');
+
+var env = require('./env.js');
+var gcloud = require('../lib')(env);
+var storage = gcloud.storage();
+
+var FILENAMES = ['bridge', 'spain', 'quit'];
+var AUDIO_FILES = {};
+var BUCKET_NAME = 'gcloud-test-bucket-temp-' + uuid.v1();
+
+FILENAMES.forEach(function(filename) {
+  var name = filename + '.raw';
+  AUDIO_FILES[filename] = {
+    name: name,
+    path: path.join(__dirname, 'data/' + name),
+    gcsUri: 'gs://' + BUCKET_NAME + '/' + name,
+    httpUri: 'https://storage.googleapis.com/' +
+      BUCKET_NAME + '/' + name
+  };
+});
+
+describe('Speech', function() {
+  var speech;
+  var bucket = storage.bucket(BUCKET_NAME);
+
+  before(function(done) {
+    async.waterfall([
+      function(next) {
+        bucket.create(next);
+      },
+      function(zone, apiResponse, next) {
+        async.map(FILENAMES, function(filename, onComplete) {
+          fs.readFile(AUDIO_FILES[filename].path, onComplete);
+        }, next);
+      },
+      function(files, next) {
+        FILENAMES.forEach(function(filename, i) {
+          AUDIO_FILES[filename].content = files[i];
+        });
+        async.map(FILENAMES, function(filename, onComplete) {
+          var file = bucket.file(AUDIO_FILES[filename].name);
+          file.save(AUDIO_FILES[filename].content, function(err) {
+            onComplete(err, file);
+          });
+        }, next);
+      },
+      function(files, next) {
+        async.map(files, function(file, onComplete) {
+          file.makePublic(onComplete);
+        }, next);
+      }
+    ], done);
+  });
+
+  beforeEach(function() {
+    speech = gcloud.speech();
+  });
+
+  after(function(done) {
+    bucket.getFiles(function(err, files) {
+      if (err) {
+        done(err);
+        return;
+      }
+
+      async.map(files, function(file, onComplete) {
+        file.delete(onComplete);
+      }, function(err) {
+        if (err) {
+          done(err);
+          return;
+        }
+
+        bucket.delete(done);
+      });
+    });
+  });
+
+  describe('recognize', function() {
+    it('recognizes speech from raw audio', function(done) {
+      fs.readFile(AUDIO_FILES.bridge.path, function(err, audioFile) {
+        assert.ifError(err);
+        speech.recognize({
+          config: {
+            encoding: 'LINEAR16',
+            sampleRate: 16000
+          },
+          audio: {
+            content: audioFile
+          }
+        }, function(err, response, apiResponse) {
```

Comment:
This feels a little confusing, having multiple response parameters. Do you think it would be better to just pass back just one of them?